src/table/components/TableTotals.tsx

```diff
@@ -16,7 +17,7 @@ function TableTotals({ rootElement, tableData, theme, layout, keyboard }) {
   return (
     <StyledHeadRow paginationNeeded={paginationNeeded} className="sn-table-row">
       {columns.map((column, columnIndex) => {
-        const cellCoord = [isTop ? 1 : rows.length + 1, columnIndex];
+        const cellCoord: [number, number] = [isTop || !rows ? 1 : rows.length + 1, columnIndex];
```

Comment:
I added more statement on rows here but I don't think I should do, it doesn't make sense to me that rows could be undefined but apparently we have a case for it in TableData interface

Reply:
Try to merge the main branch with your branch.